nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql-test/src/test/java/org/nuxeo/ecm/core/storage/sql/TestSQLBackend.java

```diff
@@ -145,6 +145,39 @@ public void testRootNode() throws Exception {
         session.close();
     }
 
+    /*
+     * The root node fragments must correctly be refetched when they are invalidated.
+     */
+    @Test
+    public void testRootNodeInvalidation() throws Exception {
+        Session session = repository.getConnection();
+        Serializable rootNodeId = session.getRootNode().getId();
+        Session session2 = repository.getConnection();
+        session2.getRootNode().getCollectionProperty(Model.ACL_PROP);
+
+        // set new acl on root
+        Node root = session.getRootNode();
+        CollectionProperty prop = root.getCollectionProperty(Model.ACL_PROP);
+        assertEquals(3, prop.getValue().length); // root acls preexist
+        ACLRow acl1 = new ACLRow(1, "test", true, "Write", "steve", null);
+        ACLRow acl2 = new ACLRow(1, "test", true, "Write", "jobs", null);
+        prop.setValue(new ACLRow[] { acl1, acl2 });
+        session.save(); // send invalidations, fragment in session2 is INVALIDATED_MODIFIED
+
+        // in session2 re-change acl
+        session2.save(); // fetch invalidations
+        Node root2 = session2.getNodeById(rootNodeId); // fetch root by id
+        CollectionProperty prop2 = root2.getCollectionProperty(Model.ACL_PROP);
+        prop2.setValue(new ACLRow[] { acl1 });
```

Comment:
Should we assert before setting new acl that we have 2 acls ? in order to assert invalidations.

Reply:
We know invalidations work in general. The last test in the method is the only special case that behaves incorrectly.